src/main/java/pages/TestHistoryDetailsPage.java

```diff
@@ -0,0 +1,44 @@
+package pages;
+
+import org.openqa.selenium.WebElement;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class TestHistoryDetailsPage extends BasePage {
+    private static final String TEST_HISTORY_DETAILS_PAGE_TITLE = "Test history details";
+    private static final String BACK_BUTTON_XPATH = "//XCUIElementTypeButton[contains(@name, 'arrow back')]";
+    private static final String PAGE_ALL_TEXT_CLASS_NAME = "XCUIElementTypeStaticText";
+
+    public void waitUntilPageIsLoaded() {
+        waitUntilPageIsLoadedById(TEST_HISTORY_DETAILS_PAGE_TITLE);
+    }
+
+    public boolean isPageTitleDisplayed() {
+        return findElementById(TEST_HISTORY_DETAILS_PAGE_TITLE).isDisplayed();
+    }
+
+    public void clickBackButton() {
+        findElementByXpath(BACK_BUTTON_XPATH).click();
+    }
+
+    public List<String> getDataSetBetweenTitles(String dataSetStartText, String dataSetEndText) {
+        List<WebElement> webElementList = findElementsByClassName(PAGE_ALL_TEXT_CLASS_NAME);
+        List<String> listOfData = new ArrayList<>();
+        boolean isInInterval = false;
```

Comment:
Why is the boolean used?

Reply:
to check if the index is in the interval or not

Reply:
it is fine